### DISCUSSION TITLE
plot(randomForestSRC.obj) produces unexpected result

### DISCUSSION THREAD

**Discussion by ar0berts** [kogalur/randomForestSRC#355] [Q&A]
Hi,

I have a [kaggle dataset](https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset) concerning stroke prediction.  when I plot the rfsrc.object I get horizontal lines for "stroke", "healthy" and "all". Thinking I  had incorrectly specified the formula or had an incorrectly constructed  the data frame I looked at the breast{randomForestSRC} dataset example:

`data(breast, package = "randomForestSRC") 

breast <- na.omit(breast) 

o <- rfsrc(status ~ ., data = breast, nsplit = 10) 

print(o) 

plot(o) `

This produces exactly the same graph suggesting  that any one of the trees is as accurate as any other without the usual progressively reducing error rate (as demonstrated by the iris classification example).

![image](https://user-images.githubusercontent.com/2455814/213903553-b1198319-7710-40cd-a9e7-69183b4ab538.png)

I would be really grateful for someone to enlighten me as to what is going on here!

BTW - by sequentially increasing ntree from 1:500 and plotting the error rate I get something more useful:

`error_plot<-data.frame(all = 0,
                       N = 0)
for(i in 1:500){
  test<-(randomForestSRC::rfsrc(status ~ ., data = na.omit(breast), nsplit = 10, ntree=i))
 error_plot[i,]<-test[["err.rate"]][i,c(1,2)]
}
error_plot.m<-reshape2::melt(error_plot)
error_plot.m$ntree<-rep(1:500,2)
ggplot2::ggplot(error_plot.m, aes(x=ntree, y=value, colour = variable)) + geom_line() + 
  labs(y="Error rate (status)", x="Number of Trees") + theme_bw()
`

![image](https://user-images.githubusercontent.com/2455814/213905202-78650982-f91f-469d-857c-697f01cb45ec.png)

**Comment by ishwaran** [comment]
By default, for computational efficiency, the error rate is computed only once for the entire forest.  From the  [rfsrc](https://www.randomforestsrc.org/reference/rfsrc.html) help file:

```
block.size: Determines how cumulative error rate is calculated.  When
          ‘NULL’, only done once for entire forest; thus plot of the
          cumulative error rate will result in a flat line.  To view
          the cumulative error rate on every nth tree, set the value to
          an integer between ‘1’ and ‘ntree’.
```

The first example from the same help file illustrates this option:

```
data(veteran, package = "randomForestSRC.stealth")
v.obj <- rfsrc(Surv(time, status) ~ ., data = veteran, ntree = 100, block.size = 1)

```

**Reply by ar0berts** [reply]
Many thanks Hemant,

Obvious now you point it out!

Andrew
